pygmt/src/velo.py

```diff
@@ -0,0 +1,252 @@
+
+
+@fmt_docstring
+@use_alias(
+    A="vector",
+    B="frame",
+    C="cmap",
+    D="rescale",
+    E="uncertainty_color",
+    G="facecolor",
+    H="scale",
+    I="intensity",
+    J="projection",
+    L="line",
+    N="no_clip",
+    R="region",
+    S="scaling",
+    U="timestamp",
+    V="verbose",
+    W="pen",
+    X="xshift",
+    Y="yshift",
+    Z="zvalue",
+    c="panel",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(R="sequence", c="sequence_comma", i="sequence_comma", p="sequence")
+def velo(self, data=None, **kwargs):
+    r"""
+    Plot velocity vectors, crosses, anisotropy bars and wedges.
+
+    Reads data values from files, numpy array or pandas DataFrame and will plot
+    the selected geodesy symbol on a map. You may choose from velocity vectors
+    and their uncertainties, rotational wedges and their uncertainties,
+    anisotropy bars, or strain crosses. Symbol fills or their outlines may be
+    colored based on constant parameters or via color lookup tables.
+
+    Must provide *data* and *scaling*.
+
+    Full option list at :gmt-docs:`supplements/geodesy/velo.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    data : str or numpy.ndarray or pandas.DataFrame
+        Either a file name, a 2D numpy array, or a pandas DataFrame with the
+        tabular data. Note that text columns are only supported with file or
+        pandas DataFrame inputs.
+
+    scaling: str
+        Selects the meaning of the columns in the data file and the figure
+        to be plotted. In all cases, the scales are in data units per length
+        unit and sizes are in length units (default length unit is controlled
+        by :gmt-term:`PROJ_LENGTH_UNIT` unless **c**, **i** , or **p** is
+        appended).
+
+        - **e**\ [*velscale*/]\ *confidence*\ [**+f**\ *font*]
+
+          Velocity ellipses in (N,E) convention. The *velscale* sets the
+          scaling of the velocity arrows. If *velscale* is not given the we
+          read it from the data file as an extra column. The *confidence* sets
+          the 2-dimensional confidence limit for the ellipse, e.g., 0.95 for
+          95% confidence ellipse. Use **+f** to set the font and size of the
+          text [Default is 9p,Helvetica,black]; give **+f**\ 0 to deactivate
+          labeling. The arrow will be drawn with the pen attributes specified
+          by the ``pen`` option and arrow-head can be colored via
+          ``facecolor``. The ellipse will be filled with the color or shade
+          specified by the ``uncertainty_color`` option [Default is
+          transparent], and its outline will be drawn if ``line`` is selected
+          using the pen selected (by ``pen`` if not given by ``line``).
+          Parameters are expected to be in the following columns:
+
+            - **1**,\ **2**: longitude, latitude of station (**-:** option
+              interchanges order)
```

Comment:
Is it better to leave out the ``**-:** option interchanges order` line, or to keep it here?

Reply:
I think best to leave it out until we add an alias for the colon option. I anticipate it would be confusing to users who do not have prior experience with command-line gmt. I added it to the list of common options that need wrapping in https://github.com/GenericMappingTools/pygmt/issues/48#issuecomment-803474203 with the possible alias 'switchcols'.

Reply:
FYI, GMT may deprecate `-:`, see https://github.com/GenericMappingTools/gmt/issues/2276

Reply:
Ok, I'll take it out then. To be honest, I don't think it's possible to do `@use_alias(:="switchcols")` either because colon is a special character in Python.